src/providerContext.tsx

```diff
@@ -1,21 +1,31 @@
 import React, { useState } from 'react'
+import DataVaultWebClient from '@rsksmart/ipfs-cpinner-client'
 
 export interface Web3ProviderContextInterface {
   provider: any | null,
-  setProvider: (value: any) => void,
+  setProvider?: (value: any) => void
+  dvClient: DataVaultWebClient | null,
+  setDvClient?: (client: DataVaultWebClient) => void
 }
 
-export const Web3ProviderContext = React.createContext<Web3ProviderContextInterface | null>(null)
+export const Web3ProviderContext = React.createContext<Web3ProviderContextInterface>({
+  provider: null,
+  dvClient: null
+})
 
 interface Web3ProviderElementInterface {
   children: React.ReactNode,
 }
 
 export const Web3ProviderElement: React.FC<Web3ProviderElementInterface> = ({ children }) => {
   const [provider, setProvider] = useState<any | null>(null)
+  const [dvClient, setDvClient] = useState<DataVaultWebClient | null>(null)
+
   const initialContext: Web3ProviderContextInterface = {
     provider: provider,
-    setProvider: (provider: any) => setProvider(provider)
+    setProvider: (provider: any) => setProvider(provider),
```

Comment:
Why `any`?

Reply:
This was from a previous PR, but if I remember correctly, I believe I used any because web3modal was using any in their example code here: https://github.com/Web3Modal/web3modal/blob/master/example/src/App.tsx#L208

None the less, I tried looking, where can I find a web3provider interface type?

Reply:
EIP-1993 😉. We can creat that type as an external package, what do you think?

Reply:
I think it is a good idea, but would this be better as an export of rLogin?